src/sqlfluff/rules/L028.py

```diff
@@ -128,8 +128,7 @@ def _generate_fixes(
 ) -> Optional[List[LintResult]]:
     """Iterate through references and check consistency."""
     # How many aliases are there? If more than one then abort.
-    assert len(table_aliases) > 0, "Only Selects with tables should be checked"
-    if len(table_aliases) > 1:
+    if len(table_aliases) != 1:
         return None
```

Comment:
Don't see why we abort on more than one, but not on 0? Same thing IMHO and this fixes one of the issues raised.